### PR TITLE
If filter matches an existing path, add it to the database

### PR DESCRIPTION
**Pull request by LinuxMercedes**

### CODE REVIEW

**Comment by euank** [comment]
I'm okay with the idea, and after thinking about it more kinda like it... but I don't think that's the right place for the code.

I think having a second function (`maybe_add_relative_to(cwd, target)` maybe?) and calling that from main in the `value_of("dir_target")` arm would make a little more sense?

**Comment by LinuxMercedes** [comment]
I firmly agree this is the wrong spot for the code; moving it elsewhere would also make it easier to turn off with a config flag if'n we end up adding that. 

I think this might be handy to have as a new-to-autojump person as I can use `z` without thinking about whether a path is in the database yet or not.

Should I change this to not visit paths that are already in the DB so we don't accidentally skew the statistics?

**Comment by euank** [comment]
If making it add-once isn't too hard, that seems like it would be better, otherwise it'll typically be an immediate double-count.

File: src/frecent_paths.rs [hunk 1]
```diff
@@ -148,6 +149,18 @@ impl PathFrecency {
             &pc_ci_sm,
         ];
 
+        // If the user has typed an exact path,
+        // add it to the database
+        env::current_dir().map(|mut cwd| {
+            cwd.push(filter);
```
Comment (euank) [inline src/frecent_paths.rs]: I assume the behaviour of `push` overwriting if `filter` is absolute is known and intended here. 
It does seem like the right behaviour.

Comment (LinuxMercedes) [reply src/frecent_paths.rs]: Yep, that was the intention!

**Comment by LinuxMercedes** [comment]
Add-once is done; it still double-counts the first time but I think that's acceptable?

File: src/main.rs [hunk 2]
```diff
@@ -177,7 +178,12 @@ fn main() {
     } else if flags.is_present("dir") {
         // Safe to unwrap because 'dir' requires 'dir_target'
         let matches = match flags.value_of("dir_target") {
-            Some(to) => frecency.directory_matches(to),
+            Some(to) => {
+                env::current_dir().map( |cwd| {
```
Comment (euank) [inline src/main.rs]: That space before `|` looks odd; did you run `cargo fmt` with a recent `rustfmt-nightly` installed?

File: src/frecency.rs [hunk 3]
```diff
@@ -57,6 +57,16 @@ where
         }
     }
 
+    pub fn insert(&mut self, key: T) {
```
Comment (euank) [inline src/frecency.rs]: frecency has decent unit test coverage; you could add a unit test for the insert-once behavior of this.

File: src/main.rs (hunk 2 above)
Comment (LinuxMercedes) [reply src/main.rs]: I put it there meself, so I shall now remove it.

**Comment by LinuxMercedes** [comment]
I'll squash this and format the resulting commit message properly once we're ready to merge!

**Comment by euank** [comment]
LGTM; I think this is good to merge once the commit messages are updated.

**Pull request merged**